src/txn/txn.c

```diff
@@ -816,6 +823,10 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	txn_global = &conn->txn_global;
 	locked = skip_update_assert = false;
 	resolved_update_count = visited_update_count = 0;
+	cursors = NULL;
+	if (txn->mod_count != 0)
+		WT_ERR(__wt_calloc(
```

Comment:
Unsure how perf sensitive this is. Is it fine for me to allocate here? Or should I make some attempt to use the stack first like we're doing here: https://github.com/wiredtiger/wiredtiger/blob/develop/src/btree/bt_ret.c#L195

Reply:
commit is definitely a performance flow function, whether to use the stack instead of heap depends on the cost of the new change? If you feel that it adds overhead, better to move it to stack.

Reply:
I have a hunch that the opening/closing of cursors + the repeated calls to `__wt_btcur_search_uncommitted` are going to dwarf this. I'm thinking of just leaving this as is unless there is an observable perf degradation after merging.

@kommiharibabu To my understanding, this strategy will still fallback to the heap for longer update lists. It will just leave enough space on the stack for the "common case", kind of like how SSO is done.

Reply:
I am also fine to go with the current change until it has major performance impact.

Reply:
This checking in `txn_commit_timestamps_assert` is only active when `assert=(...)` is configured so no customers/users will ever have that turned on. We do not want to penalize all users with the memory allocation/free for a very rare debugging case. Couldn't this allocation be done down in `txn_commit_timestamps_assert` after the return if no debugging is turned on? Then only users of the debugging feature pay the price and `cursors` will just stay NULL in the common case.

Reply:
6be3417

Only allocating in the debugging case now.